src/update-mp-v1-ids-to-v2.ts

```diff
@@ -0,0 +1,174 @@
+  }
+  return schemaPosition
+}
+
+const transform: Transform = async (file, api, _options) => {
+  if (!file.source.includes("graphql`")) {
+    return
+  }
+  const j = api.jscodeshift
+  const collection = j(file.source)
+  const schema = await promisedSchema
+
+  collection
+    .find(TaggedTemplateExpression, node => {
+      const tag = node.tag
+      return Identifier.check(tag) && tag.name === "graphql"
+    })
+    .forEach(path => {
+      const templateElement = path.node.quasi.quasis[0]
+      const graphqlDoc = parseGraphQL(templateElement.value.raw)
+
+      const newGraphQLDoc = visit(graphqlDoc, {
+        enter(node, key, parent, path, ancestors) {
+          // let p = Array.isArray(parent) ? parent[0] : parent
+          if (
+            node.kind === "Name" &&
+            parent &&
+            parent.kind == "Field" &&
+            (node.value === "id" ||
+              node.value === "_id" ||
+              node.value === "__id")
+          ) {
+            if (node.value === "__id") {
+              node.value = "id"
+              return node
+            }
+
+            if (node.value === "_id") {
+              node.value = "internalID"
+              return node
+            }
+
+            // Oh boy, it's an `id`, here we go...
+
+            let fragment = ancestors[0]
+            if (fragment && fragment.kind === "FragmentDefinition") {
+              fragment = fragment as FragmentDefinitionNode
+              const fragmentTypeName = fragment.typeCondition.name.value
+              const fragmentType = schema.getType(fragmentTypeName)
+              if (fragmentType) {
+                let parentType: ObjectTypeDefinitionNode
+
+                try {
+                  // Find the parent type of this field selection from the schema...
+                  parentType = walkSelections(
+                    path.slice(2),
+                    fragment,
+                    schema.getType(fragmentTypeName)!.astNode
+                  )
+                  // Filter down the parent type to just it's fields
+                  const v2SchemaFields = parentType.fields
+                    .filter(
+                      field =>
+                        field.name.value === "internalID" ||
+                        field.name.value === "slug"
+                    )
+                    .map(field => field.name.value)
+
+                  if (
+                    v2SchemaFields.includes("slug") &&
+                    v2SchemaFields.includes("internalID")
+                  ) {
+                    node.value = "slug"
+                    return node
+                  } else {
+                    node.value = "internalID"
+                    return node
+                  }
+                } catch {
+                  // TODO: The type wasn't found, we should inject a todo into the fragment or something
+                  node.value = "slugORinternalID"
```

Comment:
I guess doing just this is fine too, it will fail compilation and so clear draw attention.

Reply:
Yeah, leaves a clear pattern to search for.

One thing is that this codemod is removing comments generally from all the gql docs. Any idea what may be causing that?

Reply:
Yeah, generally comments aren’t part of ASTs :/ There may be a way to preserve them, but I haven’t felt the urge to really research that.